### ISSUE TITLE
Frontend project setup

### ISSUE THREAD

**Issue by g-harel** [SOEN341-Group1/KoalaCal#121]
I've elaborated the setup of our folder a little bit, here are a few points.

Please don't forget to enable/install ESlint on your code editor #116.

Also a reminder from earlier today, to start working with the project you will need to install node/npm. Once your local repo is up to date, open the command line in the frontend folder and type `npm install`. This will take a bit of time since it will be downloading all necessary modules, but it will finish eventually. To build the project once the download finishes, type `npm start` into the terminal and go to http://localhost:8080/ to see the result! This command can be left running as it will auto compile and refresh the browser window when you make changes to files.

I will start with `./app`. I have separated the files into templates and controllers. There is currently one file in each of the folders for each of the routes. We can add more routes in the future, but I've only made /login, /register and /. The template files are the html that will take over the page on the defined route. The controllers are an Angular concept, you might need to do some homework on how they work. The current code in each of the files returns a function (that you should edit) that has the `$scope` of your template as well as a reference to an API object. This object will be the source of all server side requests. If you need to add some API calls, I ask that you please add them to the API.js file and use them in your controllers afterwards. This will facilitate testing since it will act as the single portal between back and front sides of our application.

The other folder of interest is `./style`. I have also separated the css into the 3 routes and added a global app.css file. When styling your pages, please add the css into the right file and prefix your classes/ids with the route name ('.login-button' instead of '.button'). CSS is notoriously difficult to maintain on large projects, but I think these two conventions will help ours stay somewhat organized. I have also added a nifty trick to the generic css with the font-size. Instead of using `3px`, you can use `0.3rem` (divided by ten and change unit) and it will technically be the same size. BUT, if all pixel values are defined this way, it allows us to resize the whole app just by changing the `html` font-size attribute's percentage since 'rem' is a unit that represents a fraction of the html tag's font-size.

TL:DR
pls read thx 😢 

If you have any questions/problems with setting up, I should be online most of the time on slack, just send me a message.
